arcadia/aregion.cpp

```diff
@@ -1949,11 +1949,14 @@ int ARegion::CanTax(Unit *u)
 		Object *obj = (Object *) elem;
 		forlist ((&obj->units)) {
 			Unit *u2 = (Unit *) elem;
-			if (u2->guard == GUARD_GUARD && u2->IsReallyAlive())
-			    if(u2->type == U_GUARD || u2->type == U_GUARDMAGE) {
-			        if(town) return 0; //guards prevent taxing in towns
-			        if(u2->faction->ethnicity != u->faction->ethnicity) return 0; //guards prevent other ethnicities taxing
-				} else if (u2->GetAttitude(this, u) <= A_NEUTRAL) return 0;
+			if (u2->guard == GUARD_GUARD && u2->IsReallyAlive()) {
+				if (u2->type == U_GUARD || u2->type == U_GUARDMAGE) {
+					if (town) return 0; //guards prevent taxing in towns
+					if (u2->faction->ethnicity != u->faction->ethnicity) return 0; //guards prevent other ethnicities taxing
+				} else {
+					if (u2->GetAttitude(this, u) <= A_NEUTRAL) return 0;
+				}
+			}
 		}
```

Comment:
Problem here: Does this `else` (line 1956) belong to the `if` on line 1952 or the `if`` on line 1953?

following the given code:
- check for living guard (line 1952)
- check if guarding unit is an actual guard type (line 1953)
- is this region a town? -> guards + town => can not tax (return 0/false) (line1954)
- guard are NOT the same ethnicity than wannabe taxer -> can not tax (return 0/false) (line 1955)
- otherwise: are guards unfriendly? -> can not tax (line 1956)
   (otherwise related to 1 or to 1 AND 2?)
- can tax (after for-loop)

looking at the standard engine aregion.cpp function canTax():
- checks for living guard (line 1952 here)
- checks for attitude worse than neutral -> return 0 (line 1956 here)
   (false = can not tax, because unfriendly guard)
- otherwise return 1 (after for-loop)
   (true = no guard OR friendly guard)

prosposed solution:
- consider lines 1953-1955 to be additions of arcadia
- therefore match the `else` on line 1956 with the `if` on line 1952